focus/unfocus on input/textarea elements

AngeloR opened the pull request:
Pressing `f`/`F` will now display hints on input/textarea elements as well as links/buttons. Pressing the appropriate key-combo will focus on the input/textarea element. Pressing `ESC` while focused on an input or textarea element is stop focusing on that particular element (`.blur()`).

AngeloR committed:
focus/unfocus on input/textarea elements

Pressing `f`/`F` will now display hints on input/textarea elements as well as links/buttons. Pressing the appropriate key-combo will focus on the input/textarea element. Pressing `ESC` while focused on an input or textarea element is stop focusing on that particular element (`.blur()`).

AngeloR committed:
swap to using isCurrentInput method

PalmerAL commented on global.js:
```diff
@@ -103,6 +103,8 @@ function onTextTyped (key) {
         }
       } else if (link.link.tagName === 'BUTTON') {
         link.link.click()
+      } else if (isCurrentlyInInput()) {
```
why does this need to check `isCurrentlyInInput`?

AngeloR replied:
You're right this is an error - I've modified it in a more recent update.

BramVerb commented on global.js:
```diff
@@ -103,6 +103,8 @@ function onTextTyped (key) {
       } else if (link.link.tagName === 'BUTTON') {
         link.link.click()
+      } else if (isCurrentlyInInput()) {
+        link.link.focus();
```
Input elements like checkboxes and radialboxes do not change using this method. Can you add a check for such elements and simulate a `click` on them instead to toggle them.

AngeloR replied:
I've added checks for input types (checkbox/radio) and also for select elements so that you can move through the options.

AngeloR committed:
better targeting for checkbox/radio inputs

Instead of just looking at generic "inputs", we take a look at and
perform various actions based on the input type:

- checkbox: click, blur
- radio: click, blur
- select: click

For checkboxes/radios we blur immediately following the click so that we
can perform actions on the page again.

With select elements we leave the focus on the element so that a user
can use the arrow keys to select something from the dropdown.

PalmerAL commented on global.js:
```diff
@@ -103,6 +103,15 @@ function onTextTyped (key) {
         link.link.click()
+      } else if (isFocusable(link.link)) {
+        link.link.focus();
+        if(['checkbox','radio'].indexOf(link.link.attributes.getNamedItem('type').value.toLowerCase()) >= 0) {
```
Is there any reason to not just use `link.link.getAttribute('type')` here?

PalmerAL commented:
LGTM aside from the comment, @BramVerb do you see anything else?

AngeloR committed:
utilize getAttribute() call instead of attributes nodemap

PalmerAL commented:
Everything looks good to me, so I'm going to merge. Thanks for working on this!
